text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,183 @@
+
+## What alternative solutions did you consider?
+
+Instead of creating a Feature Flag CLI tool, we could expand on `cdk synth`. If a user’s feature flags differ from our recommended values, it would display a warning or suggestion. This reduces the need for another CLI command, but this method risks negatively impacting user experience due to cluttering the `cdk synth` output if many flags differ. 
+
+Another solution could be to prompt users to configure their feature flags when they create their CDK app using `cdk init`. This increases visibility to the feature flags and can be integrated smoothly within their project creation process. However, this does not help existing projects and could be considered an inconvenience and unnecessary by users. 
+
+## What is the technical solution (design) of this feature? 
+
+To implement this feature, we would first generate a Feature Flag Report artifact to be stored in CloudAssembly. The report is a `Record` of individual `FeatureFlag` objects that contain the fields `userValue`, `recommendedValue`, and `summary.` When the `cdk flags` command is run, the CLI accesses this artifact stored in `manifest.json` and displays the report to the user. A backup of the previous version of `cdk.out `will be stored in memory for future comparison. When the user chooses to modify a specific feature flag to our recommended value, that value is changed in  `cdk.json` and the entire application resynthesizes. After resynthesizing, the CloudFormation template is compared to the one prior to the change and displayed to the user. If the user wishes, they can restore their previous version by rejecting the changes. 
+![Feature Flag Advisor Design](../images/featureflagadvisor.png)
+
+
+## Additional Considerations
+
+### Increasing visibility to `cdk flags`
+
+Although this feature will increase visibility to feature flags, we still need to bring visibility to the CLI tool itself. Otherwise, CDK users continue to remain unaware of feature flags and their potential benefits. To address this, we will display a notice whenever `cdk synth` is run. This notice will be discarded once users have configured all of their flags. 
+
+```
+> cdk synth
+Notice: You currently have **10** **** unconfigured feature flags that could potentially impact
+your CDK application. Run `cdk flags` to learn more. 
+
+> cdk flags 
+    Feature Flag                              Recommended                  User
+    @aws-cdk/...                              true                         true
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - (not configured)
+  
+  MENU ----------------------------------------------------------------------
+  [1] Switch **all** flags to recommended states 
+  [2] Switch all **unconfigured** flags to recommended states
```

Comment:
I'm thinking through the user story here. Again, what is the interaction going to be if they pick this AND decide they don't like the outcome?

Do we say "revert all"?

That's fine, but what's the next step?